### PR TITLE
EOS date handling fix

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

In EOS expiration date formatting:
- there is an error branch with an exception but missing `throw`
- length condition is wrong, so if is always triggered (but doesn't do anything)

## How to test

EOS unit test

## Types of changes

* Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.

If you're adding a new blockchain

- [ ] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

**Review by hewigovens — APPROVED** [review]

File: src/EOS/Transaction.cpp [hunk 1]
```diff
@@ -99,14 +100,19 @@ void Transaction::serialize(Data& os) const noexcept{
     encodeCollection(transactionExtensions, os);
 }
 
-json Transaction::serialize() const {
+std::string Transaction::formatDate(int32_t date) {
+    // format is "2022-06-02T08:53:20", always 19 characters long
+    constexpr size_t DateSize = 19;
+    constexpr size_t BufferSize = DateSize + 1;
+    char formattedDate[BufferSize];
```
Comment (Milerius) [inline src/EOS/Transaction.cpp:107]: I think using a string stream or an std::array will make more sense here, if you use stringstream, you can use std::put_time to fill the buffer

**Comment by catenocrypt** [comment]
Some android test fails here too, but I can't see which from the output.  Maybe date conversion works differently.

**Comment by catenocrypt** [comment]
Strangely, removing of the <cassert> include fixed the Android build (even though no assert was used already)
https://github.com/trustwallet/wallet-core/pull/2275/commits/88525e61ef67c8647aafa33d6c15951c0db1eabb

**Comment by hewigovens** [comment]
Android CI might be caused by this pr: https://github.com/trustwallet/wallet-core/pull/2265, I will revert it using old ndk filter

**Comment by catenocrypt** [comment]
Android test still shows same errors, even after merging #2279 to master and here too. :(

**Comment by hewigovens** [comment]
local works tho, let me try only keeping x86

**Comment by catenocrypt** [comment]
Android build failure is sporadic.  This commit just adds an `assert`, before it CI failed, after it succeeded.  Adding the assert cannot be the reason.
https://github.com/trustwallet/wallet-core/pull/2275/commits/9a399d8ab109fc854bd07cd0c1083dad731c95ba

**Comment by catenocrypt** [comment]
The Android failure might be connected to targetSdkVersion or gradle version.

**Review by hewigovens — APPROVED** [review]

**Pull request merged**